Minio/Helper/RequestUtil.cs

```diff
@@ -0,0 +1,161 @@
+            string host = endPoint;
+            if (s3utils.IsAmazonEndPoint(endPoint))
+            {
+                // Fetch new host based on the bucket location.
+                host = AWSS3Endpoints.Instance.endpoint(region);
+                if (!usePathStyle)
+                {
+                    host = utils.UrlEncode(bucketName) + "." + utils.UrlEncode(host) + "/";
+                }
+            }
+            Uri uri = TryCreateUri(host,secure);
+            return uri;
+        }
+
+        public static Uri TryCreateUri(string endpoint,bool secure)
+        {
+            var scheme = secure ? utils.UrlEncode("https") : utils.UrlEncode("http");
+
+            // This is the actual url pointed to for all HTTP requests
+            string endpointURL = string.Format("{0}://{1}", scheme, endpoint);
+            Uri uri = null;
+            try
+            {
+                uri = new Uri(endpointURL);
+            }
+            catch (UriFormatException e)
+            {
+                throw new InvalidEndpointException(e.Message);
+            }
+            return uri;
+        }
+
+        /// <summary>
+        /// Validates URI to check if it is well formed. Otherwise cry foul.
+        /// </summary>
+        public static void ValidateEndpoint(Uri uri,string Endpoint)
+        {
+            if (string.IsNullOrEmpty(uri.OriginalString))
+            {
+                throw new InvalidEndpointException("Endpoint cannot be empty.");
+            }
+            string host = uri.Host;
+
+            if (!isValidEndpoint(uri.Host))
+            {
+                throw new InvalidEndpointException(Endpoint, "Invalid endpoint.");
+            }
+            if (!uri.AbsolutePath.Equals("/", StringComparison.CurrentCultureIgnoreCase))
+            {
+                throw new InvalidEndpointException(Endpoint, "No path allowed in endpoint.");
+            }
+
+            if (!string.IsNullOrEmpty(uri.Query))
+            {
+                throw new InvalidEndpointException(Endpoint, "No query parameter allowed in endpoint.");
+            }
+            if ((!uri.Scheme.ToLowerInvariant().Equals("https")) && (!uri.Scheme.ToLowerInvariant().Equals("http")))
+            {
+                throw new InvalidEndpointException(Endpoint, "Invalid scheme detected in endpoint.");
+            }
+            string amzHost = uri.Host;
+            if ((utils.CaseInsensitiveContains(amzHost,".amazonaws.com"))
+                 && !s3utils.IsAmazonEndPoint(uri.Host))
+            {
+                throw new InvalidEndpointException(amzHost, "For Amazon S3, host should be \'s3.amazonaws.com\' in endpoint.");
+            }
+        }
+       
+        /// <summary>
+        /// Validate Url endpoint 
+        /// </summary>
+        /// <param name="endpoint"></param>
+        /// <returns>true/false</returns>
+        public static bool isValidEndpoint(string endpoint)
+        {
```

Comment:
This function and a some others in this file appear to be duplicated in `RequestHelper.cs`. This should be avoided - there will be less code to maintain.